File: frontend/src/models/chats.js
export class Message {
    userID;
    groupID;
    private;
    content;
    constructor(content) {
        this.content = content;
    }

    getData() {
        return {
            content: this.content,
            groupID: this.groupID || -1,
            userID: this.userID || -1,
            private: this.private
        }
    }
}

File: internal/models/chats.go
package models

import "time"

type ChatList struct {
	ID      int
	Members []UserRegistration
	Name    string
	Avatar  string
}

type PrivateChat struct {
	UserID    int
	GroupID   int
	Avatar    string
	FirstName string
	LastName  string
}

type Message struct {
	ID        int
	Sender    UserRegistration
	GroupID   int
	Content   string `json:"content"`
	CreatedAt time.Time
}
